### ISSUE TITLE
tabbed page BarTextColor is not pervasive and can't be applied after instantiation

### ISSUE THREAD

**Issue by PunkStarStudios** [xamarin/Xamarin.Forms#1323]
### Description
The barcode text setting for Tabbed Pages does not "stick" after instantiating. So if you add/remove children from the tabbed page, that text colour won't apply to the new children, and attempts to reapply that setting are in vain.

### Steps to Reproduce

1.  Create Tabbed Page with multiple children
2. Set BarTextColor to White
3. Remove child in code, add a new one.
4. Observe new children will not have white bartextcolor.

XAML would look like :
```
   <TabbedPage.BarBackgroundColor>
       <Color x:Uid="colorPrimary">#61a60e</Color>
   </TabbedPage.BarBackgroundColor>

   <TabbedPage.BarTextColor>
       <Color x:Uid="colorPrimary">#ffffff</Color>
   </TabbedPage.BarTextColor>

   <TabbedPage.BackgroundColor>
       <Color x:Uid="colorPrimary">#61a60e</Color>
   </TabbedPage.BackgroundColor>

   <TabbedPage.Children>
       <local:HomePageSavings Title="Savings" x:Name="btnSavings" Icon="Home.png" IsVisible="True"/>
       <local:LoyaltyCard Title="Loyalty" x:Name="btnLoyalty" Icon="Loyalty.png"  IsVisible="True"/>
       <local:eFlyer Title="eFlyer" x:Name="btneFlyer" Icon="eFlyer.png"  IsVisible="True"/>
       <local:GiftCard Title="Giftcard" x:Name="btnGift" Icon="Gift.png"  IsVisible="True"/>
   </TabbedPage.Children>
```
Now in C# code, try swapping out some pages with (MainPage and LinkLoyalty being content view pages):

               this.Children.RemoveAt(0);
               this.Children.Insert(0, new MainPage { Title = "Home", Icon = "Home.png" });

               this.Children.RemoveAt(1);
               this.Children.Insert(1, new LinkLoyalty { Title = "Loyalty", Icon = "Loyalty.png" });

Try applying the setting again with something like this.BarBackgroundColor = Color.White.

### Expected Behavior
White should be the text color of all the child tabs. Alternatively, c# code to set the text color after adding a child should change the text color... but it doesn't.

### Actual Behavior
You'll see that it won't apply. The untouched tab still has white text, but any children added do not.

### Basic Information

Xamarin Forms v. 2.5

### Reproduction Link

https://github.com/xamarin/Xamarin.Forms/files/1535251/Bug1323.zip

**Comment by mohachouch** [comment]
Which platforms are affected by this bug?

**Comment by PunkStarStudios** [comment]
Sorry.
Looks to be iOS and Windows.
But Android does something else entirely... just not sure if it's by design or if it's a bug.
On Android, all the text is white as expected.

But the icon for the selected tab on the Android doesn't change colour at all (while they do with the other two platforms to indicate the selected tab).

**Comment by StephaneDelcroix** [comment]
the Xaml sample is not valid. XF Xaml Parser does not, AFAIK, supports `x:Uid`

**Comment by jassmith** [comment]
Can you please attach a reproduction project?

**Comment by PunkStarStudios** [comment]
I have no idea why you say the XAML is not valid. It compiles and runs completely fine for me (albeit with the text color bug)... unless I am misunderstanding what you are saying. But, never the less, I'll create a scaled down sample project and submit it this morning.

I should mention, that I had brought this up to Rob Caplan (Rob.Caplan@microsoft.com) who confirmed that this is indeed a bug (he was able to reproduce it) and offered to create the bug report if I didn't want to.

I'll add the sample project shortly....

**Comment by StephaneDelcroix** [comment]
@PunkStarStudios at the very least, the `x:Uid` is ignored. I can't remind right now if it fails or not on unhandled `x:` attributes

**Comment by PunkStarStudios** [comment]
I've created a sample project. But the zip is massive (even though it's basically skeleton code).
What folders/files can I safely remove from the zip before uploading for you all?

**Comment by PunkStarStudios** [comment]
Need mind... found the instructions on what to omit... will be uploading shortly...

**Comment by PunkStarStudios** [comment]
Actually - the instructions on what to submit are not accurate. "Clean project and delete packages/bin/obj folders before sending"... there is no such folder. So I took a while guess and cleaned a bunch out. If this doesn't work for people... let me know what I have to include in the ZIP and I'll re-upload.
[Bug1323.zip](https://github.com/xamarin/Xamarin.Forms/files/1535251/Bug1323.zip)

**Comment by rmarinho** [comment]
fixed by #2109
